File: backend/app/Models/Survey.ts
import { DateTime } from 'luxon'
import { BaseModel, BelongsTo, belongsTo, column } from '@ioc:Adonis/Lucid/Orm'
import User from './User'

export default class Survey extends BaseModel {
  @column({ isPrimary: true })
  public id: number

  @column()
  public user_id: number

  @column()
  public name: string
  
  @column()
  public link: string

  @column.dateTime({ autoCreate: true })
  public createdAt: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  public updatedAt: DateTime

  @belongsTo(() => User, {
    localKey: "user_id"
  })
  public user: BelongsTo<typeof User>
}

File: frontend/lib/models/survey.dart
class Survey {
  String link, name;
  int id;

  Survey(this.link, this.name);

  Survey.fromJson(Map<String, dynamic> json)
      : link = json['link'],
        name = json['name'],
        id = json['id'];

  Map<String, dynamic> toJson() => {
        'link': link,
        'name': name,
      };
}
